ImageFeed/Services/NetworkService.swift

```diff
@@ -0,0 +1,64 @@
+//
+//  NetworkService.swift
+//  ImageFeed
+//
+//  Created by Егор Свистушкин on 30.03.2023.
+//
+
+import Foundation
+
+// MARK: - Network Error
+enum NetworkError: Error {
+    case httpStatusCode(Int)
+    case urlRequestError(Error)
+    case urlSessionError
+}
+
+// MARK: - HTTP Request
+
+extension URLRequest {
+    static func makeHTTPRequest(path: String, httpMethod: String, baseURL: URL = defaultBaseURL) -> URLRequest {
+        var request = URLRequest(url: URL(string: path, relativeTo: baseURL)!)
+        request.httpMethod = httpMethod
+        return request
+    }
+    
+    mutating func setToken(_ token: String) {
+        setValue("Bearer \(token)", forHTTPHeaderField: "Authorization")
+    }
+}
+
+// MARK: - Network Connection
+
+extension URLSession {
+    
+    //Добавил здесь дженерик 4.04.23, 11:52
+    func data <Data: Decodable>   (for request: URLRequest, completion: @escaping (Result<Data, Error>) -> Void) -> URLSessionTask {
```

Comment:
Пробелы лишние, принято вот так:
```suggestion
    func data<Data: Decodable>(for request: URLRequest, completion: @escaping (Result<Data, Error>) -> Void) -> URLSessionTask {
```